### PR TITLE
Move remappers inside atm procs

### PR DESCRIPTION
**Pull request by bartgol**
Addresses #936.

### CODE REVIEW

**Comment by bartgol** [comment]
The branch passes tests on my machine. However, there's one to do item, that I'm not sure whether should be in this PR or not.

I ended up changing Homme, to have all inputs being just the states. This is in line with the idea that Homme will back out tendencies. However, I haven't added logic to back them out (needs local copies of initial states, e.g). Should I add it here? Should I revert the changes so that Homme still uses FM,FQ,FT?

I'm tempted to leave the branch as-is. The next step is to make Homme do the right things for inputs/outputs. In particular, it should back out tendencies, as well as do T<->vTheta_dp conversions (in and out).

Note: one reason to remove FM, FT, and FQ as inputs is related to #941: the fields were going to be provided as `Field<const Real>` to Homme (since they're inputs), which prevented me from registering them inside a remapper (which, as things are now, requires both src and tgt to be non-const fields).

File: components/scream/src/share/atm_process/atmosphere_process_group.cpp [hunk 1]
```diff
@@ -86,30 +86,13 @@ AtmosphereProcessGroup (const ekat::Comm& comm, const ekat::ParameterList& param
...
-  //    proper AbstractRemapper implementation for the reference-actual
-  //    grids pair. The APG stores internally a bunch of remappers,
-  //    which calls during the `run` call, to perform the remappings.
-
-  // Resizing the remappers vectors
-  m_inputs_remappers.resize(m_group_size);
-  m_outputs_remappers.resize(m_group_size);
+  // There is a catch when it comes to exposing a list of required/computed
+  // fields: if (and only if) the splitting is sequential, and contains
+  // 2+ processes, it is possible that some inputs of the non-first atm
```
Comment (jeff-cohere) [inline components/scream/src/share/atm_process/atmosphere_process_group.cpp:91]: Maybe rephrase: "..., it is possible that some inputs of atmosphere processes in the list are computed by processes that precede them."

File: components/scream/src/dynamics/homme/atmosphere_dynamics.cpp [hunk 2]
```diff
@@ -161,156 +146,70 @@ set_updated_group (const FieldGroup<Real>& group)
...
-  const int num_elems = state.num_elems();
-  const int num_tracers = tracers.num_tracers();
-  using Scalar = Homme::Scalar;
+  // Print homme's parameters, so user can see whether something wasn't set right.
+  ::Homme::Context::singleton().get<::Homme::SimulationParams>().print();
 
-  constexpr int NTL  = HOMMEXX_NUM_TIME_LEVELS;
-  constexpr int NVL  = HOMMEXX_NUM_LEV;
-  constexpr int NVLI = HOMMEXX_NUM_LEV_P;
+  // Now that all fields on the ref grid are set, create the fields on the
```
Comment (jeff-cohere) [inline components/scream/src/dynamics/homme/atmosphere_dynamics.cpp:161]: Is this just a leftover comment?

File: components/scream/src/dynamics/homme/atmosphere_dynamics.cpp [hunk 3]
```diff
@@ -161,156 +146,70 @@ set_updated_group (const FieldGroup<Real>& group)
...
-      auto ps_in = f.template get_reshaped_view<const Real*[NTL][NP][NP]>();
-      using ps_type = std::remove_reference<decltype(ps)>::type;
-      auto non_const_ptr = const_cast<Real*>(ps_in.data());
-      ps = ps_type(non_const_ptr,num_elems);
-    } else {
-      ekat::error::runtime_abort("Error! Unexpected field name. This is an internal error. Please, contact developers.\n");
-    }
-  }
+  // Finish homme initialization
+  // Homme::initialize_dp3d_from_ps_c();
```
Comment (jeff-cohere) [inline components/scream/src/dynamics/homme/atmosphere_dynamics.cpp:168]: Is this work still to be done?

**Review by jeff-cohere — APPROVED** [review]
This looks much simpler from the perspective of atmosphere processes, groups, and the DAG.

I think it's fine to back out the tendencies in Homme in another PR.

**Comment by AaronDonahue** [comment]
I concur with Jeff, I think it would be okay to break the backing out of tendencies into a separate PR.

File: components/scream/src/share/atm_process/atmosphere_process_group.cpp (hunk 1 above)
Comment (bartgol) [reply components/scream/src/share/atm_process/atmosphere_process_group.cpp:91]: Neat. Thanks!

File: components/scream/src/dynamics/homme/atmosphere_dynamics.cpp (hunk 2 above)
Comment (bartgol) [reply components/scream/src/dynamics/homme/atmosphere_dynamics.cpp:161]: Yup.

File: components/scream/src/dynamics/homme/atmosphere_dynamics.cpp (hunk 3 above)
Comment (bartgol) [reply components/scream/src/dynamics/homme/atmosphere_dynamics.cpp:168]: Ah, no. So, standalone Homme (from e3sm) does init dp3d from ps and from reference pressure levels thickness. However, in order to pick up ps from the initial conditions file, I would have had to expose ps as an input for HommeDynamics (otherwise it would not be given to Homme). Furthermore, even if I did that, `dp` would still need an I.C., according to the AD, since it's an input for the atm. Which would be confusing: you ask for an I.C. for `dp`, but it seems to not be refleted in the numbers (since the code will replace that with `dp=some_func(ps,ref_vert_coord)`.

I _think_ the easiest thing to do is to forget about `dp=some_func(ps,ref_vert_coord)`, and simply init `dp` with what's in the I.C. file. If you guys agree, I'll simply remove the commented line.

**Comment by bartgol** [comment]
Sorry Aaron, I thought I asked also for your review, but I guess I didn't click correctly (or clicked twice).

File: components/scream/src/dynamics/homme/atmosphere_dynamics.cpp (hunk 3 above)
Comment (jeff-cohere) [reply components/scream/src/dynamics/homme/atmosphere_dynamics.cpp:166]: That sounds okay to me. What do you think, @AaronDonahue ?

**Review by jeff-cohere — COMMENTED** [review]
Looks good to me, @bartgol . I'll wait to see what @AaronDonahue thinks about removing the comment related to `dp`'s initial conditions.

File: components/scream/src/dynamics/homme/atmosphere_dynamics.cpp (hunk 3 above)
Comment (AaronDonahue) [reply components/scream/src/dynamics/homme/atmosphere_dynamics.cpp:166]: I concur.  I'm guessing there will be some code cleanup in the future to address things like this.  For example if we decide it really doesn't make sense to include `dp` as an init'd variable.  But right now out goal is to get the code coupled and running.  So we can certainly revisit in the future if warranted.

**Review by jeff-cohere — APPROVED** [review]

**Comment by bartgol** [comment]
> Looks good to me, @bartgol . I'll wait to see what @AaronDonahue thinks about removing the comment related to `dp`'s initial conditions.

Ah, shoot, I should have not added the automerge label then. Well, we can still think about dp I.C..

**Comment by jeff-cohere** [comment]
> > Looks good to me, @bartgol . I'll wait to see what @AaronDonahue thinks about removing the comment related to `dp`'s initial conditions.
> 
> Ah, shoot, I should have not added the automerge label then. Well, we can still think about dp I.C..

Aaron already weighed in, so I think this PR is set.

**Comment by bartgol** [comment]
Ah, I missed that. Great.
